test/support/query.ex

```diff
@@ -0,0 +1,5 @@
+defmodule Authsense.Test.Query do
+  alias Authsense.Test.Repo
+
+  def where(_model, extra_field: extra_field), do: Enum.filter(Repo.all(), &(&1.extra_field == extra_field))
```

Comment:
:warning: Also, `scope` should be a `Queryable`, not the result of `Repo.all`.

That is:

In fact, if you were to let `scope` pass through in the implementation of `get_scope/1`, you can directly:

```elixir
Service.authenticate(scope: (fn -> query end))
```

In fact, if you let non-function `scope` args pass through with `get_scope/1`, you can simply:

```elixir
|> Service.authenticate(scope: query)
```

Much much neater ;)